File: app/src/main/java/com/sdex/activityrunner/db/history/HistoryModelDao.java
package com.sdex.activityrunner.db.history;

import android.arch.lifecycle.LiveData;
import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;
import java.util.List;

@Dao
public interface HistoryModelDao {

  @Insert(onConflict = OnConflictStrategy.REPLACE)
  void insert(HistoryModel... model);

  @Update(onConflict = OnConflictStrategy.REPLACE)
  void update(HistoryModel... models);

  @Delete
  void delete(HistoryModel... models);

  @Query("SELECT * FROM HistoryModel ORDER BY id DESC")
  LiveData<List<HistoryModel>> getHistory();

  @Query("DELETE FROM HistoryModel")
  void clean();
}

File: app/src/main/java/com/sdex/activityrunner/db/history/HistoryModelDao.kt
package com.sdex.activityrunner.db.history

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update
import kotlinx.coroutines.flow.Flow

@Dao
interface HistoryModelDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insert(vararg model: HistoryModel)

    @Update(onConflict = OnConflictStrategy.REPLACE)
    suspend fun update(vararg models: HistoryModel)

    @Delete
    suspend fun delete(vararg models: HistoryModel)

    @Query("SELECT * FROM HistoryModel ORDER BY id DESC")
    fun getHistory(): Flow<List<HistoryModel>>

    @Query("DELETE FROM HistoryModel")
    suspend fun clean()
}
